### PR TITLE
docs: translate useState page

### PR DESCRIPTION
**Pull request by vashafaris**
Closes https://github.com/reactjs/id.react.dev/issues/409

Description
Translate useState page.
Page URL: [useState](https://id.react.dev/reference/react/useState)

### CODE REVIEW

File: src/content/reference/react/useState.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ title: useState
 
 <Intro>
 
-`useState` is a React Hook that lets you add a [state variable](/learn/state-a-components-memory) to your component.
+`useState` merupakan React Hook yang memungkinkan Anda untuk menambahkan [variabel *state*](/learn/state-a-components-memory) pada komponen anda.
```
Comment (mazipan) [inline src/content/reference/react/useState.md:7]: "Anda", always use "A" capital.

File: src/content/reference/react/useState.md [hunk 2]
```diff
@@ -32,32 +32,32 @@ function MyComponent() {
   // ...
 ```
 
-The convention is to name state variables like `[something, setSomething]` using [array destructuring.](https://javascript.info/destructuring-assignment)
+Konvensi dalam menamai variable *state* adalah menggunakan pola `[something, setSomething]` dengan [*array destructuring*.](https://javascript.info/destructuring-assignment)
 
-[See more examples below.](#usage)
+[Lihat contoh lainnya di bawah ini.](#usage)
```
Comment (mazipan) [inline src/content/reference/react/useState.md:37]:
```suggestion
[Lihat lebih banyak contoh di bawah ini.](#usage)
```

File: src/content/reference/react/useState.md [hunk 3]
```diff
@@ -32,32 +32,32 @@ function MyComponent() {
...
 
-#### Returns {/*returns*/}
+#### Kembalian {/*returns*/}
 
-`useState` returns an array with exactly two values:
+`useState` mengembalikan sebuah senarai yang memiliki dua nilai:
 
-1. The current state. During the first render, it will match the `initialState` you have passed.
-2. The [`set` function](#setstate) that lets you update the state to a different value and trigger a re-render.
+1. *State* saat ini. Pada saat pertama kali *render*, itu akan sama dengan `initialState` yang anda oper sebelumnya.
```
Comment (mazipan) [inline src/content/reference/react/useState.md:48]: Anda, with A capital

Comment (mazipan) [inline src/content/reference/react/useState.md:48]: Anda, with A capital

File: src/content/reference/react/useState.md (hunk 1 above)
Comment (zainfathoni) [reply src/content/reference/react/useState.md:7]: [Reference](https://ejaan.kemdikbud.go.id/eyd/penggunaan-huruf/huruf-kapital/#23-huruf-kapital-digunakan-sebagai-huruf-pertama-kata-penunjuk-hubungan-kekerabatan-seperti-bapak-ibu-kakak-dan-adik-serta-kata-atau-ungkapan-lain-termasuk-unsur-bentuk-ulang-utuh-yang-digunakan-sebagai-sapaan):

image

Comment (vashafaris) [reply src/content/reference/react/useState.md]: Thank you for the reference!

**Review by mazipan — APPROVED** [review]

**Comment by mazipan** [comment]
Asking for a peer review @zainfathoni @r17x @mhaidarhanif @resir014

File: src/content/reference/react/useState.md [hunk 4]
```diff
@@ -68,34 +68,34 @@ function handleClick() {
...
-* In Strict Mode, React will **call your updater function twice** in order to [help you find accidental impurities.](#my-initializer-or-updater-function-runs-twice) This is development-only behavior and does not affect production. If your updater function is pure (as it should be), this should not affect the behavior. The result from one of the calls will be ignored.
+* Pada mode ketat, React akan **memanggil fungsi pembaruan Anda dua kali** untuk [membantu Anda menemukan kejadian yang tidak diharapkan.](#my-initializer-or-updater-function-runs-twice) Ini hanya terjadi di lingkungan pengembangan dan tidak memengaruhi produksi. Jika fungsi pembaruan Anda murni (seperti seharusnya), ini tidak akan memengaruhi perilakunya. Hasil dari salah satu panggilan akan diabaikan.
 
 ---
 
-## Usage {/*usage*/}
+## Penggunaan {/*usage*/}
 
-### Adding state to a component {/*adding-state-to-a-component*/}
+### Menambahkan *state* pada sebuah komponen {/*adding-state-to-a-component*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:96]:
```suggestion
### Menambahkan state pada sebuah komponen {/*adding-state-to-a-component*/}
```

Don't italicise loanwords in headings.

https://github.com/reactjs/id.react.dev/wiki/Panduan-Penulisan-Umum#kata-serapan-di-dalam-heading

File: src/content/reference/react/useState.md [hunk 5]
```diff
@@ -168,9 +168,9 @@ export default function Counter() {
 
 <Solution />
 
-#### Text field (string) {/*text-field-string*/}
+#### *Text field* (*string*) {/*text-field-string*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:171]:
```suggestion
#### Text field (string) {/*text-field-string*/}
```

File: src/content/reference/react/useState.md [hunk 6]
```diff
@@ -200,9 +200,9 @@ export default function MyInput() {
 
 <Solution />
 
-#### Checkbox (boolean) {/*checkbox-boolean*/}
+#### *Checkbox* (*boolean*) {/*checkbox-boolean*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:203]:
```suggestion
#### Checkbox (boolean) {/*checkbox-boolean*/}
```

File: src/content/reference/react/useState.md [hunk 7]
```diff
@@ -236,9 +236,9 @@ export default function MyCheckbox() {
 
 <Solution />
 
-#### Form (two variables) {/*form-two-variables*/}
+#### *Form* (dua variabel) {/*form-two-variables*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:239]:
```suggestion
#### Form (dua variabel) {/*form-two-variables*/}
```

File: src/content/reference/react/useState.md [hunk 8]
```diff
@@ -276,9 +276,9 @@ button { display: block; margin-top: 10px; }
 
 ---
 
-### Updating state based on the previous state {/*updating-state-based-on-the-previous-state*/}
+### Memperbarui *state* berdasarkan *state* sebelumnya {/*updating-state-based-on-the-previous-state*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:279]:
```suggestion
### Memperbarui state berdasarkan state sebelumnya {/*updating-state-based-on-the-previous-state*/}
```

File: src/content/reference/react/useState.md [hunk 9]
```diff
@@ -416,16 +416,16 @@ h1 { display: block; margin: 10px; }
 
 ---
 
-### Updating objects and arrays in state {/*updating-objects-and-arrays-in-state*/}
+### Memperbarui objek dan senarai di dalam *state*. {/*updating-objects-and-arrays-in-state*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:419]:
```suggestion
### Memperbarui objek dan senarai di dalam state {/*updating-objects-and-arrays-in-state*/}
```

File: src/content/reference/react/useState.md [hunk 10]
```diff
@@ -435,13 +435,13 @@ setForm({
 });
 ```
 
-Read [updating objects in state](/learn/updating-objects-in-state) and [updating arrays in state](/learn/updating-arrays-in-state) to learn more.
+Baca [memperbarui objek pada *state*](/learn/updating-objects-in-state) dan [memperbarui senarai pada *state](/learn/updating-arrays-in-state) untuk belajar lebih lanjut.
 
 <Recipes titleText="Examples of objects and arrays in state" titleId="examples-objects">
 
-#### Form (object) {/*form-object*/}
+#### *Form* (objek) {/*form-object*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:442]:
```suggestion
#### Form (objek) {/*form-object*/}
```

File: src/content/reference/react/useState.md [hunk 11]
```diff
@@ -512,9 +512,9 @@ input { margin-left: 5px; }
 
 <Solution />
 
-#### Form (nested object) {/*form-nested-object*/}
+#### *Form* (objek bersarang) {/*form-nested-object*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:515]:
```suggestion
#### Form (objek bersarang) {/*form-nested-object*/}
```

File: src/content/reference/react/useState.md [hunk 12]
```diff
@@ -882,35 +882,35 @@ function ItemList({ artworks, onToggle }) {
 
 ---
 
-### Avoiding recreating the initial state {/*avoiding-recreating-the-initial-state*/}
+### Menghindari membuat ulang *state* awal {/*avoiding-recreating-the-initial-state*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:885]:
```suggestion
### Menghindari membuat ulang state awal {/*avoiding-recreating-the-initial-state*/}
```

File: src/content/reference/react/useState.md [hunk 13]
```diff
@@ -961,9 +961,9 @@ export default function TodoList() {
 
 <Solution />
 
-#### Passing the initial state directly {/*passing-the-initial-state-directly*/}
+#### Mengoper *state* awal secara langsung {/*passing-the-initial-state-directly*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:964]:
```suggestion
#### Mengoper state awal secara langsung {/*passing-the-initial-state-directly*/}
```

File: src/content/reference/react/useState.md [hunk 14]
```diff
@@ -1018,13 +1018,13 @@ export default function TodoList() {
 
 ---
 
-### Resetting state with a key {/*resetting-state-with-a-key*/}
+### Mereset *state* menggunakan kunci (*key*) {/*resetting-state-with-a-key*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:1021]:
```suggestion
### Menyetel ulang state menggunakan kunci (key) {/*resetting-state-with-a-key*/}
```

reset -> setel ulang

File: src/content/reference/react/useState.md [hunk 15]
```diff
@@ -1069,27 +1069,27 @@ button { display: block; margin-bottom: 20px; }
 
 ---
 
-### Storing information from previous renders {/*storing-information-from-previous-renders*/}
+### Menyimpan informasi dari *render* sebelumnya {/*storing-information-from-previous-renders*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:1072]:
```suggestion
### Menyimpan informasi dari render sebelumnya {/*storing-information-from-previous-renders*/}
```

File: src/content/reference/react/useState.md [hunk 16]
```diff
@@ -1138,17 +1138,17 @@ button { margin-bottom: 10px; }
...
+Pola ini bisa sulit dipahami dan biasanya lebih baik untuk dihindari. Namun, ini lebih baik daripada memperbarui *state* dalam efek. Ketika Anda memanggil fungsi `set` selama *render*, React akan me-*render* ulang komponen tersebut sesaat setelah komponen Anda keluar dengan pernyataan `return`, dan sebelum me-*render* anak-anak. Dengan begitu, anak-anak tidak perlu me-*render* dua kali. Sisa fungsi komponen Anda tetap akan dieksekusi (dan hasilnya akan dibuang). Jika kondisi Anda berada di bawah semua panggilan Hook, Anda dapat menambahkan `return` lebih awal untuk memulai ulang *render* sebelumnya.
 
 ---
 
-## Troubleshooting {/*troubleshooting*/}
+## Pemecahan Masalah {/*troubleshooting*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:1147]:
```suggestion
## Pemecahan masalah {/*troubleshooting*/}
```

Use sentence case for heading, as per global site styleguide.

File: src/content/reference/react/useState.md [hunk 17]
```diff
@@ -1177,16 +1177,16 @@ console.log(nextCount); // 1
 
 ---
 
-### I've updated the state, but the screen doesn't update {/*ive-updated-the-state-but-the-screen-doesnt-update*/}
+### Saya telah memperbarui *state*, tetapi tidak ada perbaruan pada layar {/*ive-updated-the-state-but-the-screen-doesnt-update*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:1180]:
```suggestion
### Saya telah memperbarui state, tetapi tidak ada perbaruan pada layar {/*ive-updated-the-state-but-the-screen-doesnt-update*/}
```

File: src/content/reference/react/useState.md [hunk 18]
```diff
@@ -1138,17 +1138,17 @@ button { margin-bottom: 10px; }
...
+Pola ini bisa sulit dipahami dan biasanya lebih baik untuk dihindari. Namun, ini lebih baik daripada memperbarui *state* dalam efek. Ketika Anda memanggil fungsi `set` selama *render*, React akan me-*render* ulang komponen tersebut sesaat setelah komponen Anda keluar dengan pernyataan `return`, dan sebelum me-*render* anak-anak. Dengan begitu, anak-anak tidak perlu me-*render* dua kali. Sisa fungsi komponen Anda tetap akan dieksekusi (dan hasilnya akan dibuang). Jika kondisi Anda berada di bawah semua panggilan Hook, Anda dapat menambahkan `return` lebih awal untuk memulai ulang *render* sebelumnya.
 
 ---
 
-## Troubleshooting {/*troubleshooting*/}
+## Pemecahan Masalah {/*troubleshooting*/}
 
-### I've updated the state, but logging gives me the old value {/*ive-updated-the-state-but-logging-gives-me-the-old-value*/}
+### Saya sudah memperbarui *state*, tetapi saat saya mencetak log nilainya masih yang lama {/*ive-updated-the-state-but-logging-gives-me-the-old-value*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:1149]:
```suggestion
### Saya sudah memperbarui state, tetapi saat saya mencetak log nilainya masih yang lama {/*ive-updated-the-state-but-logging-gives-me-the-old-value*/}
```

File: src/content/reference/react/useState.md [hunk 19]
```diff
@@ -1198,9 +1198,9 @@ setObj({
 
 ---
 
-### I'm getting an error: "Too many re-renders" {/*im-getting-an-error-too-many-re-renders*/}
+### Saya mendapatkan galat: *"Too many re-renders"* {/*im-getting-an-error-too-many-re-renders*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:1201]:
```suggestion
### Saya mendapatkan galat: "Too many re-renders" {/*im-getting-an-error-too-many-re-renders*/}
```

File: src/content/reference/react/useState.md [hunk 20]
```diff
@@ -1213,13 +1213,13 @@ return <button onClick={handleClick}>Click me</button>
...
-If you can't find the cause of this error, click on the arrow next to the error in the console and look through the JavaScript stack to find the specific `set` function call responsible for the error.
+Jika Anda tidak dapat menemukan penyebab kesalahan ini, klik pada panah di sebelah galat pada konsol dan lihat melalui tumpukan JavaScript untuk menemukan panggilan fungsi `set` yang bertanggung jawab atas kesalahan tersebut.
 
 ---
 
-### My initializer or updater function runs twice {/*my-initializer-or-updater-function-runs-twice*/}
+### Fungsi inisialisasi atau pembaruan saya berjalan dua kali {/*my-initializer-or-updater-function-runs-twice*/}
 
-In [Strict Mode](/reference/react/StrictMode), React will call some of your functions twice instead of once:
+Pada [*Strict Mod*e*](/reference/react/StrictMode), React akan memanggil beberapa fungsi Anda sebanyak dua kali alih-alih hanya satu kali:
```
Comment (resir014) [inline src/content/reference/react/useState.md:1222]:
```suggestion
Pada [*Strict Mode*](/reference/react/StrictMode), React akan memanggil beberapa fungsi Anda sebanyak dua kali alih-alih hanya satu kali:
```

File: src/content/reference/react/useState.md [hunk 21]
```diff
@@ -1261,15 +1261,15 @@ setTodos(prevTodos => {
...
-Now that this updater function is pure, calling it an extra time doesn't make a difference in behavior. This is why React calling it twice helps you find mistakes. **Only component, initializer, and updater functions need to be pure.** Event handlers don't need to be pure, so React will never call your event handlers twice.
+Apabila fungsi pembaruan ini murni, memanggilnya beberapa kali tidak membuat perbedaan perilaku. Inilah mengapa React memanggilnya dua kali untuk membantu Anda menemukan kesalahan. **Hanya fungsi komponen, inisialisasi, dan pembaruan yang perlu bersifat murni.** *Event handler* tidak perlu murni, sehingga React tidak akan pernah memanggil *event handler* Anda dua kali.
 
-Read [keeping components pure](/learn/keeping-components-pure) to learn more.
+Baca [menjaga komponen murni](/learn/keeping-components-pure) untuk belajar lebih lanjut.
 
 ---
 
-### I'm trying to set state to a function, but it gets called instead {/*im-trying-to-set-state-to-a-function-but-it-gets-called-instead*/}
+### Saya mencoba mengatur *state* ke sebuah fungsi, namun malah dipanggil {/*im-trying-to-set-state-to-a-function-but-it-gets-called-instead*/}
```
Comment (resir014) [inline src/content/reference/react/useState.md:1270]:
```suggestion
### Saya mencoba mengatur state ke sebuah fungsi, namun malah dipanggil {/*im-trying-to-set-state-to-a-function-but-it-gets-called-instead*/}
```

**Review by resir014 — CHANGES_REQUESTED** [review]
Some requested change. @vashafaris

**Comment by vashafaris** [comment]
Thanks for the feedback! Updated, please rereview @resir014

**Review by resir014 — APPROVED** [review]
All good now. Thanks!

**Pull request merged**